include/onnxruntime/core/framework/op_kernel_context.h

```diff
@@ -26,6 +26,14 @@ class OpKernelContext {
   MLDataType InputType(int index) const;
   MLDataType OutputType(int index) const;
 
+  const OrtValue* GetInputOrtValue(int index) const {
+    return GetInputMLValue(index);
+  }
+
+  OrtValue* GetOutputOrtValue(int index) {
+    return GetOrCreateOutputMLValue(index);
+  }
+
   template <typename T>
```

Comment:
nit: the existing OpKernelContext interface carefully provides input/output methods that enforce various constraints and hide the underlying OrtValue. adding something generic like this, above the specific methods, with no comment loudly calling out that the vast majority of code should never ever get the OrtValue this way, is likely to lead to places uses these methods that shouldn't.

Can we name these something very specific so their intended usage is clearer? 

Also what is the expected behavior if the OrtValue is nullptr? Should that be tested here or does the calling code need to be aware of how/when that could happen so it can handle it (these methods need doco explaining this if that's the case)?

Reply:
I would rather add methods that specific to Optional.

Reply:
@skottmckay : This method goes away now